### PR TITLE
EPMRPP-25318 cucumber-js agent

### PR DESCRIPTION
**Pull request by Kayuaga**
Cucumber-js agent .

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,244 @@
...
+}
+```
+
+3. Create a main launch file as in example below:
+
+#### cuceLaunch.js
+```javascript
+'use strict'
+const { spawn } = require('child_process');
+const config = {
```
Comment (lexecon) [inline README.md]: use require('./reportportalConf')

File: README.md [hunk 2]
```diff
@@ -1 +1,244 @@
...
+4. Then create or refactor handlers.js file by adding ID  as in example below:
+
+#### handlers.js file
+
+```javascript
+
+const {CucumberReportPortalHandler} = require('../../../../modules');
+var {defineSupportCode} = require('cucumber');
+defineSupportCode(consumer => CucumberReportPortalHandler({
+ token: "00000000-0000-0000-0000-000000000000",
```
Comment (lexecon) [inline README.md]: use require('./reportportalConf')

File: README.md [hunk 3]
```diff
@@ -1 +1,244 @@
-# agent-js-cucumber
\ No newline at end of file
+# agent-js-cucumber
+### This cucumber agent works well with cucumber version < 3.0.0
+Install agent to your project dir
+<code>npm install agent-js-cucumber --save-dev </code>
```
Comment (lexecon) [inline README.md]: \```cmd
 npm install agent-js-cucumber --save-dev
\```

File: package.json [hunk 4]
```diff
@@ -0,0 +1,36 @@
+{
+  "name": "agent-js-cucumber",
+  "version": "1.0.0",
```
Comment (lexecon) [inline package.json]: use 0.1.0

File: package.json [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+    "log4js": "^2.3.3",
+    "selenium-webdriver": "^3.5.0"
+  },
+  "scripts": {
+    "test": "cd testSample && node cuceLaunch.js",
+    "testSingle": "cd testSample && npm run testSingle  "
+  },
+  "repository": {
+    "type": "git",
+    "url": "git+https://github.com/Kayuaga/agent-js-cucumber.git"
```
Comment (lexecon) [inline package.json]: use report portal repository

File: package.json [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+  },
+  "keywords": [
+    "cucumber-reporter",
+    "reportportal",
+    "reporter"
+  ],
+  "author": "Uladzimir Paliakou",
+  "license": "ISC",
+  "bugs": {
+    "url": "https://github.com/Kayuaga/agent-js-cucumber/issues"
```
Comment (lexecon) [inline package.json]: use report portal repository

File: package.json [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+  },
+  "homepage": "https://github.com/Kayuaga/agent-js-cucumber#readme"
```
Comment (lexecon) [inline package.json]: use report portal repository

File: testSample/cuceLaunch.js [hunk 8]
```diff
@@ -0,0 +1,44 @@
+'use strict'
+const {spawn} = require('child_process');
+const config = {
```
Comment (lexecon) [inline testSample/cuceLaunch.js]: use require(config)

File: testSample/features/step_definitions/support/handlers.js [hunk 9]
```diff
@@ -0,0 +1,13 @@
+const {CucumberReportPortalHandler} = require('../../../../modules');
+var {defineSupportCode} = require('cucumber');
+defineSupportCode(consumer => CucumberReportPortalHandler({
```
Comment (lexecon) [inline testSample/features/step_definitions/support/handlers.js]: use require(config)

File: test.js [hunk 10]
```diff
@@ -0,0 +1,4 @@
+let {CucumberReportPortalHandler} = require('./index.js');
+
+
+console.deb('Vovan')
```
Comment (dshlychkou) [inline test.js]: It looks like it is 'debug' file. Pls remove it from PR or explain why you need it.

File: testSample/features/par2.feature [hunk 11]
```diff
@@ -0,0 +1,7 @@
+Feature: Logging two
```
Comment (dshlychkou) [inline testSample/features/par2.feature]: Pls consider to add tests for:
Attachments (png, json, txt) send to RP as log message
Send screenshots as log message
Run scenarios with 'Scenario Outlines'
Run scenarios where steps receive Tables
Run scenarios with 'Example' part
Run scenario where step is failed, skipped, undefined

File: modules/cucumber-epam-reportportal-handler.js [hunk 12]
```diff
@@ -0,0 +1,356 @@
...
+                    let errorMessage = `${event.stepDefinition.uri}\n ${util.format(event.failureException)}`//`${event.stepDefinition.uri}\n ${event.failureException}`;
+                    reportportal.log({
+                        item_id: context.stepId,
+                        time: reportportal._now(),
+                        level: "ERROR",
+                        message: errorMessage
+                    })
+                        .then(result => {
+                            if(config.takeScreenshot === 'onFailure'){
+                                browser.takeScreenshot()
```
Comment (dshlychkou) [inline modules/cucumber-epam-reportportal-handler.js]: This is going to work for Protractor. Pls add test for this.

File: test.js (hunk 10 above)
Comment (Kayuaga) [reply test.js]: removed

**Pull request merged**